### DISCUSSION TITLE
Estrutura do Projeto

### DISCUSSION THREAD

**Discussion by rosajsx** [he4rt/react4noobs#104] [General]
Este tópico de discussão tem o intuito de elencarmos os tópicos necessários e obrigatórios para o projeto.
A partir dessa discussão poderemos construir a tabela de projetos.

Precisamos pensar principalmente na:
- Estrutura do Repositório
- Conteúdos relevantes
- O que faz sentido trazer ou não, por exemplo: Explicar detalhes do NextJS, ou alguma framework por exemplo.

Gostaria de citar inicialmente que gostaria que concluíssemos o necessário para se trabalhar em React como escopo do "Pronto pra consumo". E coisas como bibliotecas utilitários e outros como um "Plus". O que acham?

**Comment by Rychillie** [comment]
Eu acho que devemos focar em separar server-components, ainda é cedo falarmos sobre pois não existem muitas empresas utilizando o mesmo ainda.

Na minha visão o Server-Components veio para a Vercel vender seus serviços, então deveriamos trazer ele, porem não focar inteiramente.

Acredito que deveriamos focar mais nas funcionalidades do React em si, o que o usuario vai usar e podemos até trazer algumas Libs, desde gerenciamento de estado, até estilização.

Temos aqui uma oportunidade de criar o maior repositório a ser usado pelo mundo todo.

**Reply by rosajsx** [reply]
Concordo demais! Acredito que poderia ter uma seção apenas pra ele, como um extra. (Depois das funcionalidades principais)!

**Comment by MatheusFilg** [comment]
Acredito que no caso do Next o que pode ser feito é mostrar a versão 12 e 13, porém tendo um enfoque maior no 12 já que acredito que é onde tem maior utilização. 
Sobre a estrutura do repositório acredito que esteja ok e que devemos apenas acrescentar os conteúdos que faltam, que ao meu ver alguns deles são: as versões do next 12 & 13, Tailwinds, Redux Toolkit, Zustand, Electron e entre outros que estão listados como "em breve"

**Reply by rosajsx** [reply]
Assim, NextJS é praticamente uma tecnologia a parte. Utiliza o React porém tem muita funcionalidade pronta. Tu não acha que talvez apenas falar sobre e direcionar a um 4noobs de next, seria mais interessante não?.

Concordo sobre Redux, Zustand. 
Electron eu enxergo com um extra, fora do conteudo principal

**Reply by MatheusFilg** [reply]
Concordo, é uma boa estrutura sim!

**Comment by LimaoCode** [comment]
#### Muitas plataformas e cursos onlines fazem spam de novas libs e tecnologias relacionadas ao React, na minha visão o nosso repo deveria ensinar conceitos sólidos para fortalecer a base, isso facilitará a entrada no mercado pois ele terá que lidar com legado e grande parte das vezes não vai encontrar essas novas tec's nesse tipo de projeto.

Um bom norte seria usar o roadmap.sh como base e filtrar a partir dele:
https://roadmap.sh/react

![roadmap/react](https://github.com/he4rt/react4noobs/assets/34945812/d0c341e0-9134-49d5-b670-3e562f9ead37)


#### Diante desses pontos acho de extrema importância ressaltar a necessidade de conhecimento em JavaScript para se aprender o React, evidenciar o CRA e a explicar por que a adesão do Vite, citar Class Component e a mudança para Functional Components e dar foco para assuntos como ciclo de vida,render props, JSX, Hooks e demais conceituais que formam a base sólida.

**Reply by rosajsx** [reply]
Concordo demais! Acho que esse Roadmap vai ser um ótimo exemplo pra utilizarmos nas novas estruturas.

**Comment by rosajsx** [comment]
Bom galera, acredito que deu pra tirar bastante coisa! Vamos seguir agora com planos de ação! Vou criar um Projeto no repositório para podermos ir andando com tudo!

**Comment by rosajsx** [comment]
Primos, criei um board no projeto para que possamos ir organizando as tasks direitinho.
Sintam-se livres para pegar task, adicionar, remover. Bora sempre tentar estar conversando entre si e mandar ver!
https://github.com/he4rt/react4noobs/projects/2
